### PR TITLE
DM-36648: add num_samples to DeviceConfig

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

**Review by wvreeven — APPROVED** [review]
Looks good to me.

**Review by wvreeven — CHANGES_REQUESTED** [review]
Sorry, I approved too soon. See my comments to the ts_ess_csc PR.

**Review by wvreeven — APPROVED** [review]
Thanks for the discussion on Slack and taking away my concerns regarding the right project in which to put the code in question.

File: python/lsst/ts/ess/common/device_config.py [hunk 1]
```diff
@@ -55,6 +58,7 @@ def __init__(
         baud_rate: int,
         location: str,
         num_channels: int = 0,
+        num_samples: int = 25,
```
[the remark this answers is not in the document]
Comment (r-owen) [reply python/lsst/ts/ess/common/device_config.py:61]: Good point. I will initialize it to 0. I also plan to make this config a dataclass.

**Pull request merged**